web/src/store/dandiset.ts

```diff
@@ -84,24 +89,27 @@ const dandisetModule = defineModule({
       commit.setVersions(null);
       commit.setOwners(null);
       commit.setLoading(false);
+      commit.setVersionCount(0);
     },
     async initializeDandisets(context: any, { identifier, version }) {
       const { dispatch } = dandisetActionContext(context);
       await dispatch.uninitializeDandisets();
 
-      // this can be done concurrently, don't await
-      dispatch.fetchDandisetVersions({ identifier });
+      // first fetch all versions, set total count
+      // use count to fetch latest dandiset, must be sequential
+      await dispatch.fetchDandisetVersions({ identifier });
       await dispatch.fetchDandiset({ identifier, version });
       await dispatch.fetchOwners(identifier);
```

Comment:
I realize this would involve a backend change, but I think the way to properly handle this would be to add a sort parameter to the version list endpoint, which would allow us to sort versions by most recent, and just choose the first one from there (removing the need to call `mostRecentVersion` from within `fetchDandiset`. We could then store that in a variable in the store as well (named something like `mostRecentVersion`.)

I think it would be within scope of this PR to add that, but if you would rather just get this merged we can punt on that and file an issue.

Reply:
I agree, sorting would be a more straightforward way of getting the most recent version. I don't mind continuing the work here itself. Meanwhile, I will move this to draft.